### ISSUE TITLE
`query` hook preventing build

### ISSUE THREAD

**Issue by carlos-menezes** [openmultiplayer/launcher#42]
The `query` hook is making the build fail,

```sh
src/hooks/query.tsx:31:7 - error TS2322: Type 'Timeout' is not assignable to type 'number'.
```

**Comment by AmyrAhmady** [comment]
I had to explicitly use number because of errors I was getting for using Timeout
So both my machine (node 16) and CI (node 20) are working fine without spitting out errors. I already knew this but still ran CI against your PR just to double check and here are the results:

https://github.com/openmultiplayer/launcher/actions/runs/6907382025/job/18794514620?pr=43

The issue could be your linter, maybe, because timers are from web standards, they are numbers, not Node's implementation of it

Either way, can you tell me which version of node you're using?

**Comment by ksenonadv** [comment]
> I had to explicitly use number because of errors I was getting for using Timeout So both my machine (node 16) and CI (node 20) are working fine without spitting out errors. I already knew this but still ran CI against your PR just to double check and here are the results:
> 
> https://github.com/openmultiplayer/launcher/actions/runs/6907382025/job/18794514620?pr=43
> 
> The issue could be your linter, maybe, because timers are from web standards, they are numbers, not Node's implementation of it
> 
> Either way, can you tell me which version of node you're using?

Use ReturnType<typeof setInterval>

**Comment by carlos-menezes** [comment]
@AmyrAhmady thought I had replied to your comment. I'm using Node 20.

```sh
➜  ~ node -v            
v20.5.1
```
Likewise, I'm getting errors when compiling the tauri counterpart of this project:

```sh
error[E0425]: cannot find function `get_nickname` in module `samp`
error[E0599]: `std::option::Option<std::string::String>` doesn't implement `std::fmt::Display`
```

This might warrant its own issue though.

**Comment by AmyrAhmady** [comment]
Both of these issues should be fixed now in [14d930d2a7bc02fec6484eaf593ec42e73d9ca3d ](https://github.com/openmultiplayer/launcher/commit/14d930d2a7bc02fec6484eaf593ec42e73d9ca3d) and https://github.com/openmultiplayer/launcher/commit/881cd365de7fd37904ac3d94a18e6857908c7252
